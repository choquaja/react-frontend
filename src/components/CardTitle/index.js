import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { themeColor } from '../../services/theme';

const isInverted = props => props.invert && css`
  margin: -1rem -1rem 1rem;
  padding: 1rem;
  background: ${themeColor('black')};
  color: ${themeColor('white')};
`;

const CardTitle = styled.h1`
  font-size: 1.6em;
  margin: 0 0 1rem;
  ${isInverted}
`;

CardTitle.propTypes = {
  invert: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

CardTitle.defaultProps = {
  invert: false,
};

export default CardTitle;
