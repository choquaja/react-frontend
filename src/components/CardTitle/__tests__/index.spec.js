import React from 'react';
import { shallow } from 'enzyme';
import CardTitle from '../';

const required = {
  children: <div>Test</div>,
};
describe('CardTitle', () => {
// Smoke test
  it('Renders without crashing', () => {
    shallow(<CardTitle {...required} />);
  });

// Other Tests
  it('Acts appropriately when given prop invert', () => {
    const component = shallow(<CardTitle {...required} invert />);
    expect(component).toMatchSnapshot();
  });
});
