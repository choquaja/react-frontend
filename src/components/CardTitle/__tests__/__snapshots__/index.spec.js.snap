// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle Acts appropriately when given prop invert 1`] = `
<h1
  className="CardTitle__CardTitle-bwxVvc kkeJLu"
>
  <div>
    Test
  </div>
</h1>
`;
